packages/@aws-cdk/service-spec-build/src/resource-builder.ts

```diff
@@ -11,27 +11,41 @@ import {
   TypeDefinition,
 } from '@aws-cdk/service-spec-types';
 
+/**
+ * Options for the resourceBuilder API
+ */
+export interface ResourceBuilderOptions {
+  description?: string;
+  region?: string;
+  primaryIdentifier?: string[];
+}
+
 /**
  * Adds resources and types to a spec database
  */
 export class SpecBuilder {
   constructor(public readonly db: SpecDatabase) {}
 
-  public resourceBuilder(typeName: string, options: { description?: string; region?: string } = {}) {
+  public resourceBuilder(typeName: string, options: ResourceBuilderOptions = {}) {
     const existing = this.db.lookup('resource', 'cloudFormationType', 'equals', typeName);
 
     if (existing.length > 0) {
       const resource = existing.only();
       if (!resource.documentation && options.description) {
         resource.documentation = options.description;
       }
+      if (!resource.primaryIdentifier) {
+        resource.primaryIdentifier = options.primaryIdentifier;
+      }
```

Comment:
I think this means that we are importing the old spec first and then the new spec, so most resources already exist in the database. Can someone confirm whether that assumption is right? Is there an equivalent property that I should be adding to the `primaryIdentifier` field from the old spec?

Reply:
You are correct, and the old spec doesn't have this information.

All good!